File: Source/Fayde.Client/Typescript/Controls/ToolTip.ts
/// <reference path="ContentControl.ts" />
/// CODE
/// <reference path="Enums.ts" />

module Fayde.Controls {
    export class ToolTip extends ContentControl {
        static HorizontalOffsetProperty: DependencyProperty = DependencyProperty.Register("HorizontalOffset", () => Number, ToolTip);
        static VerticalOffsetProperty: DependencyProperty = DependencyProperty.Register("VerticalOffset", () => Number, ToolTip);
        static IsOpenProperty: DependencyProperty = DependencyProperty.Register("IsOpen", () => Boolean, ToolTip);
        static PlacementProperty: DependencyProperty = DependencyProperty.Register("Placement", () => new Enum(PlacementMode), ToolTip);
        static PlacementTargetProperty: DependencyProperty = DependencyProperty.Register("PlacementTarget", () => UIElement, ToolTip);
        HorizontalOffset: Number;
        VerticalOffset: Number;
        IsOpen: boolean;
        Placement: PlacementMode;
        PlacementTarget: UIElement;
    }
    Fayde.RegisterType(ToolTip, {
    	Name: "ToolTip",
    	Namespace: "Fayde.Controls",
    	XmlNamespace: Fayde.XMLNS
    });
}

File: Source/Fayde.Server/Controls/Tooltip.cs
using Fayde.Core;
using Fayde.Xaml.Metadata;

namespace Fayde.Controls
{
    public class ToolTip : ContentControl
    {
        public static readonly PropertyDescription Placement = PropertyDescription.Register("Placement", typeof(PlacementMode), typeof(ToolTip));
        public static readonly PropertyDescription PlacementTarget = PropertyDescription.Register("PlacementTarget", typeof(UIElement), typeof(ToolTip));
    }
}